src/librustdoc/test.rs

```diff
@@ -727,7 +758,53 @@ impl Tester for Collector {
                     linker,
                     edition,
                     persist_doctests
-                )
+                );
+
+                if let Err(err) = res {
+                    match err {
+                        TestFailure::CompileError => (), // The compiler errors are enough
```

Comment:
I think there should still be a message like "Couldn't compile the test." in this case.